frontend/src/components/taskSelection/map.js

```diff
@@ -0,0 +1,166 @@
+import React, { useLayoutEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { extent } from 'geojson-bounds';
+import mapboxgl from 'mapbox-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+
+import { MAPBOX_TOKEN } from '../../config';
+import lock from '../../assets/img/lock.png';
+
+mapboxgl.accessToken = MAPBOX_TOKEN;
+
+export const TasksMap = ({
+  mapResults,
+  className,
+  projectId,
+}) => {
+  const mapRef = React.createRef();
+  const [map, setMapObj] = useState(null);
+  const activeTasks = useSelector(state => state.tasks.get('activeTasks'));
+  const activeProject = useSelector(state => state.tasks.get('activeProject'));
+  const dispatch = useDispatch();
+
+  useLayoutEffect(() => {
+    /* May be able to refactor this to just take
+     * advantage of useRef instead inside other useLayoutEffect() */
+    /* I referenced this initially https://philipprost.com/how-to-use-mapbox-gl-with-react-functional-component/ */
+    if (MAPBOX_TOKEN) {
+      setMapObj(
+        new mapboxgl.Map({
+          container: mapRef.current,
+          // style: 'mapbox://styles/mapbox/bright-v9',
+          style: 'mapbox://styles/mapbox/bright-v9',
+          zoom: 0,
+        }),
+      );
+    }
+
+    return () => {
+      map && map.remove();
+    };
+    // eslint-disable-next-line
+  }, []);
+
+  useLayoutEffect(() => {
+    /* docs: https://docs.mapbox.com/mapbox-gl-js/example/cluster/ */
+    const mapboxLayerDefn = () => {
+      map.addSource('tasks', {
+        type: 'geojson',
+        data: mapResults,
+      });
+      const lockIcon = new Image(17, 20);
+      lockIcon.src = lock;
+      map.addImage('lock', lockIcon, {width: 17, height: 20, data: lockIcon})
+
+      map.addLayer({
+        id: 'tasks-icon',
+        type: 'symbol',
+        source: 'tasks',
+        layout: {
+          'icon-image': [
+            'match',
+            ['get', 'taskStatus'],
+            'LOCKED_FOR_MAPPING', 'lock',
+            'LOCKED_FOR_VALIDATION', 'lock',
+            ''
+          ],
+          'icon-size': 0.7
+        }
+      });
+      map.addLayer({
+        id: 'tasks-fill',
+        type: 'fill',
+        source: 'tasks',
+        paint: {
+          'fill-color': [
+            'match',
+            ['get', 'taskStatus'],
+            'READY', '#fff',
+            'LOCKED_FOR_MAPPING', '#fff',
+            'MAPPED', '#a1d7e5',
+            'LOCKED_FOR_VALIDATION', '#a1d7e5',
+            'VALIDATED', '#6cb570',
+            'INVALIDATED', '#e6e6e6',
+            'BADIMAGERY', '#e04141',
+            'rgba(0,0,0,0)'
+          ],
+        }
+      }, 'tasks-icon');
+      map.addLayer({
+        id: 'selected-tasks-border',
+        type: 'line',
+        source: 'tasks',
+        paint: {
```

Comment:
I think the task borders should be smaller at wider zoom levels so it doesn't turn all white. Sometimes it is easy to design/test those things on a throw-away map in Mapbox Studio and then download the final zip file of the MapboxGL styles from the main page to copy/paste into the Javascript. (edited)